### ISSUE TITLE
[Bug] NeoForge 1.20.4 - Infinite BlackScreen on Load

### ISSUE THREAD

**Issue by HalestormXV** [BloCamLimb/ModernUI#212]
**Describe the bug**
Infinite BlackScreen on Load. 

**To Reproduce**
1. Started my Modpack
2. Let Mod Loader start
3. Gets to Mojang Load Screen
4. Stays at a BlackScreen. No Crash, no error, no exit code, just a BlackScreen

**Expected behavior**
To Load to the Main Menu

**Environment**
 - Modern UI 3.10.1.5 (NeoForge)
 - NeoForge Version: 20.4.232
 - Other Mods: https://modrinth.com/modpack/journey-beyond/version/1.0.3
 
 I read the Description Notes. I know you said the mod replaced functionality of Controlling so I removed it during this test, and you also states that if you were using ImmediatelyFast to disable some of the settings which was done according to the instructions. However the BlackScreen issue still occurred. Now in the compatibility section you do mention OptiFine, Sodium, Iris, Create, AoA3. I use AoA3 in this pack and I use Embeddium which is as you know a Fork of Sodium. I am not sure if that is causing this BlackScreen issue but perhaps if compatibility with Sodium exists then compatibility with Embeddium will/should exist? It is also important to note that I use FancyMenu as well as the DrippyLoading Screen addon. So I see my Mojang custom loading screen and it gets to the half-way point before the BlackScreen issue occurs. Alternatively it should also be noted that I am using ResourcePacks, although I don't forsee how they would cause an issue with the Mod itself starting the game as I cant even get to the Main Menu. The moment I disable this mod everything loads with no issues and I can play as normal, so I know it is this mod but I am unsure what the incompatibility might be as I did follow those instructions.

**Comment by BloCamLimb** [comment]
We need to know your GPU, its driver version, OS version, and a full `debug.log` in the last run, which is located in your `[instance or .minecraft] / logs` directory.

**Comment by HalestormXV** [comment]
> I have noticed that some things prevent resource loading from finishing, resulting in `Minecraft.gameLoadFinished` being false for a long time or forever. In this case, Minecraft will skip rendering, so you will see a black screen.
> 
> But after three hours of testing, I couldn't find which mod changed the resource loading behavior, sometimes the black screen doesn't occur. But at least I can say that ModernUI won't affect this. And this is not caused by any of the mods you mentioned: Controlling ImmediatelyFast Embeddium FancyMenu DrippyLoading ResourcefulLib.
> 
> I think you have to spend more time filtering out which mod or minimal set of mods can reproduce this.
> 
> By the way, do not use Emojiful as it will cause the game to freeze during loading, as it performs network I/O on the main thread. It can also break the text engine of ModernUI and finally crash the game. ModernUI is capable rendering Unicode emoji inside a Unicode text layout with small CPU&memory usage.
> 
> ![image](https://private-user-images.githubusercontent.com/14106648/326243346-0d963d4f-aec1-4b70-a486-2f91a61a1f75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzMTQwNDUsIm5iZiI6MTcxNDMxMzc0NSwicGF0aCI6Ii8xNDEwNjY0OC8zMjYyNDMzNDYtMGQ5NjNkNGYtYWVjMS00YjcwLWE0ODYtMmY5MWE2MWExZjc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI4VDE0MTU0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4MTVhNjA4MjEyMDg1ZWZjOWU3YjhhNDUwOGU1Mjg4YmY4ZmYzYjc2MmY4ZmQzMDE5OGJhNDc1MTZhODdhZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JY6RA5BCU1CMvBmn8eyZa3PGpK2W_I-ZWxy_3GllbcI)

I believe your your very first sentence might have helped me narrow it down. I will do some further testing but I do have a Mod in here that overrides Resources. ResourcePackOverrides specifically. However, if I am understanding correctly it could mean that not only could that mod potentially be causing an issue (because it does have an option to attempt to reload resources on a failed load) but that the various ResourcePacks in general could be contributing to the issue as one of them could be causing this Minecraft.gameLoadFinished being false.

I will do some further testing and see if I can find a result as I have a Development version of this pack that I can mess around on without affecting my production line. I will then re-post here with the result so even if it isn't the fault of this Mod, perhaps a note can be placed for future users.

**Comment by HalestormXV** [comment]
After some very long testing. I have seemed to have narrowed it down to these 4 mods. If any of these 5 mods are active the Black Screen returns.

- yet-another-config-lib-neoforge-3.3.2+1.20.4.jar
- craftedcore-3.2.4-neoforge.jar
- remorphed-3.6-neoforge.jar
- seamless-loading-screen-2.1.1+1.20.4-neoforge.jar
- walkers-4.2-neoforge.jar

Once I narrowed it down I was consecutively able to start up the pack and get to the Main Menu Screen with those mods disabled. However then I started loading in some of my previously disabled mods and the Black Screen returned even with those 5 disabled. So my guess is whatever mods do something similar to those 5 will cause the Black Load Screen and "broken resource issue." Not sure if this is helpful, or perhaps you might know what these 5 mods do that I can then work around and figure out what other mods might do similar things.

**Comment by BloCamLimb** [comment]
I can't point out anything suspicious. Some mods use architectury.

**Comment by BloCamLimb** [comment]
Any news?

**Comment by HalestormXV** [comment]
I was away for a few days. I will be reworking on this later this evening (in about 8 or so hours my time) for some time to see what I can come up with. And reply/edit this post. I saved my check points so I believe it will be uncovered.

**Comment by HalestormXV** [comment]
It would appear after some testing I managed to get it to now work with the pack. I disabled the mod CleanTooltips, ClientTweaks and Controlling and updated the Balm library mod. I am not exactly sure which one of those may have done it. Perhaps Balm since it was just recently updated from another issue that I reported to them a week or so ago, or maybe the ClientTweaks/CleanTooltips combo. I am not entirely sure but it seems to be working properly now.
